src/PactNet.Native/Scenarios.cs

```diff
@@ -0,0 +1,26 @@
+using System;
+using System.Collections.Generic;
+
+namespace PactNet.Native
+{
+    /// <summary>
+    /// Defines the scenarios static list for messaging support
+    /// </summary>
+    public class Scenarios
+    {
+        /// <summary>
+        /// The available scenarios
+        /// </summary>
+        internal static Dictionary<string, Func<dynamic>> All;
```

Comment:
This is a global static field and is not thread safe. The consumer tests (at least for request/response) may run interactions in different threads in parallel, and I think that means you'd have major problems with this approach.

Reply:
I forgot the new Dictionary<string, Func<dynamic>>(); which should be thread-safe per microsoft.
The best would be to test it, improve the sample app to add two classes with diff provider tests. I'll see to add it.